rtt_tf/rtt_tf-component.hpp

```diff
@@ -2,36 +2,44 @@
 #define OROCOS_RTT_TF_COMPONENT_HPP
 
 #include <rtt/RTT.hpp>
-#include <tf/tf.h>
-#include <tf/tfMessage.h>
+#include <tf2_msgs/TFMessage.h>
+#include <tf2/buffer_core.h>
 
 namespace rtt_tf
 {
   // Inherit from TaskContext and Transformer, the second is required in order to use tf_prefix
-  class RTT_TF: public RTT::TaskContext, protected tf::Transformer
+  class RTT_TF: public RTT::TaskContext, protected tf2::BufferCore
   {
     static const int DEFAULT_BUFFER_SIZE = 100;
 
-    boost::shared_ptr<tf::Transformer> m_transformer;
     double prop_cache_time;
     double prop_buffer_size;
     std::string prop_tf_prefix;
 
-    RTT::InputPort<tf::tfMessage> port_tf_in;
-    RTT::OutputPort<tf::tfMessage> port_tf_out;
+    RTT::InputPort<tf2_msgs::TFMessage> port_tf_in;
+    RTT::InputPort<tf2_msgs::TFMessage> port_tf_static_in;
+    RTT::OutputPort<tf2_msgs::TFMessage> port_tf_out;
+
+    void internalUpdate(
+        tf2_msgs::TFMessage& msg,
+        RTT::InputPort<tf2_msgs::TFMessage>& port);
+
+    ros::Time getCommonTime(
```

Comment:
Rename to `getLatestCommonTime()` for consistency with [tf2::BufferCore::_getLatestCommonTime()](http://docs.ros.org/kinetic/api/tf2/html/classtf2_1_1BufferCore.html#af20f2a5ad43089c8cd3db1b04badb194)?